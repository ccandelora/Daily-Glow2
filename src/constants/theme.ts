export const COLORS = {
  // Primary colors for emotions (keeping these for emotional context)
  primary: {
    red: '#FF6B6B',     // Warm red for negative emotions
    green: '#00C853',   // Vibrant green for positive emotions
    blue: '#2979FF',    // Bright blue for calm emotions
    yellow: '#FFD600',  // Bright yellow for energetic emotions
  },
  
  // UI colors updated for dark theme with glowing accents
  ui: {
    background: 'rgba(28, 14, 45, 0.95)',  // Dark purple background
    card: 'rgba(38, 20, 60, 0.85)',        // Slightly lighter purple for cards
    text: '#FFFFFF',                        // White text for contrast
    textSecondary: 'rgba(255, 255, 255, 0.7)', // Semi-transparent white
    border: 'rgba(82, 67, 194, 0.3)',      // Subtle purple border
    disabled: 'rgba(82, 67, 194, 0.2)',    // Dimmed purple for disabled state
    accent: '#4169E1',                     // Royal blue for accents
    highlight: 'rgba(65, 105, 225, 0.15)', // Soft blue highlight
  },

  // Status colors with glow effects
  status: {
    success: '#00E676',
    error: '#FF5252',
    warning: '#FFD600',
    info: '#4169E1',
  },

  // Gradient colors for glow effects
  gradient: {
    start: 'rgba(65, 105, 225, 0.2)',  // Royal blue glow
    middle: 'rgba(147, 112, 219, 0.15)', // Purple glow
    end: 'rgba(28, 14, 45, 0.9)',     // Dark purple base
  },
};

export const FONTS = {
  sizes: {
    xs: 12,
    sm: 14,
    md: 16,
    lg: 22,
    xl: 28,
    xxl: 36,
    xxxl: 44,
  },
  weights: {
    regular: '400' as const,
    medium: '500' as const,
    semibold: '600' as const,
    bold: '700' as const,
  },
  // Adding font families for more sophistication
  families: {
    heading: 'System',
    body: 'System',
  }
};

export const SPACING = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 40,
};

export const BORDER_RADIUS = {
  sm: 4,
  md: 8,
  lg: 16,
  xl: 24,
  circle: 999,
};

export const TIME_PERIODS = {
  MORNING: {
    label: 'Morning',
    range: { start: 5, end: 11 }, // 5:00 AM - 11:59 AM
    greeting: 'Good morning',
    icon: 'üåÖ'
  },
  AFTERNOON: {
    label: 'Afternoon',
    range: { start: 12, end: 16 }, // 12:00 PM - 4:59 PM
    greeting: 'Good afternoon',
    icon: '‚òÄÔ∏è'
  },
  EVENING: {
    label: 'Evening',
    range: { start: 17, end: 4 }, // 5:00 PM - 4:59 AM
    greeting: 'Good evening',
    icon: 'ÔøΩÔøΩ'
  }
} as const;

export default {
  COLORS,
  FONTS,
  SPACING,
  BORDER_RADIUS,
  TIME_PERIODS,
}; 