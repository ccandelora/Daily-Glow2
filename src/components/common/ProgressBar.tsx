import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface ProgressBarProps {
  progress: number; // 0 to 1
  color?: string;
  style?: ViewStyle;
  height?: number;
}

export const ProgressBar: React.FC<ProgressBarProps> = ({
  progress,
  color = theme.COLORS.primary.green,
  style,
  height = 4,
}) => {
  const validProgress = Math.min(Math.max(progress, 0), 1);

  return (
    <View style={[styles.container, { height }, style]}>
      <View
        style={[
          styles.progress,
          {
            width: `${validProgress * 100}%`,
            backgroundColor: color,
            height,
          },
        ]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    backgroundColor: theme.COLORS.ui.border,
    borderRadius: theme.BORDER_RADIUS.circle,
    overflow: 'hidden',
  },
  progress: {
    height: '100%',
    borderRadius: theme.BORDER_RADIUS.circle,
  },
}); 