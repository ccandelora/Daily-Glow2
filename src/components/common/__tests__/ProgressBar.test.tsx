import React from 'react';
import { render } from '@testing-library/react-native';
import { ProgressBar } from '../ProgressBar';
import theme from '@/constants/theme';
import { View } from 'react-native';

describe('ProgressBar Component', () => {
  test('renders correctly with default props', () => {
    const { toJSON } = render(<ProgressBar progress={0.5} />);
    expect(toJSON()).toMatchSnapshot('default-progress-bar');
  });

  test('applies correct progress width', () => {
    const testCases = [
      { progress: 0, expectedWidth: '0%' },
      { progress: 0.25, expectedWidth: '25%' },
      { progress: 0.5, expectedWidth: '50%' },
      { progress: 0.75, expectedWidth: '75%' },
      { progress: 1, expectedWidth: '100%' },
    ];

    testCases.forEach(({ progress, expectedWidth }) => {
      const { toJSON } = render(<ProgressBar progress={progress} />);
      expect(toJSON()).toMatchSnapshot(`progress-bar-width-${expectedWidth}`);
    });
  });

  test('clamps progress values outside 0-1 range', () => {
    // Test with progress < 0
    const { toJSON: toJSONNegative } = render(<ProgressBar progress={-0.5} />);
    expect(toJSONNegative()).toMatchSnapshot('progress-bar-negative-value');
    
    // Test with progress > 1
    const { toJSON: toJSONOver } = render(<ProgressBar progress={1.5} />);
    expect(toJSONOver()).toMatchSnapshot('progress-bar-over-value');
  });

  test('applies custom color', () => {
    const customColor = '#FF0000';
    const { toJSON } = render(<ProgressBar progress={0.5} color={customColor} />);
    expect(toJSON()).toMatchSnapshot(`progress-bar-color-${customColor}`);
  });

  test('applies custom height', () => {
    const customHeight = 8;
    const { toJSON } = render(<ProgressBar progress={0.5} height={customHeight} />);
    expect(toJSON()).toMatchSnapshot(`progress-bar-height-${customHeight}`);
  });

  test('applies custom style', () => {
    const customStyle = { marginTop: 10, marginBottom: 10 };
    const { toJSON } = render(<ProgressBar progress={0.5} style={customStyle} />);
    expect(toJSON()).toMatchSnapshot('progress-bar-custom-style');
  });
}); 