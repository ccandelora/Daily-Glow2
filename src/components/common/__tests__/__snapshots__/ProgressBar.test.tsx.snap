// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component applies correct progress width: progress-bar-width-0% 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "0%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies correct progress width: progress-bar-width-25% 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "25%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies correct progress width: progress-bar-width-50% 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "50%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies correct progress width: progress-bar-width-75% 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "75%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies correct progress width: progress-bar-width-100% 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies custom color: progress-bar-color-#FF0000 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#FF0000",
          "height": 4,
          "width": "50%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies custom height: progress-bar-height-8 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 8,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 8,
          "width": "50%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component applies custom style: progress-bar-custom-style 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      {
        "marginBottom": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "50%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component clamps progress values outside 0-1 range: progress-bar-negative-value 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "0%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component clamps progress values outside 0-1 range: progress-bar-over-value 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`ProgressBar Component renders correctly with default props: default-progress-bar 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(82, 67, 194, 0.3)",
        "borderRadius": 999,
        "overflow": "hidden",
        "width": "100%",
      },
      {
        "height": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "borderRadius": 999,
          "height": "100%",
        },
        {
          "backgroundColor": "#00C853",
          "height": 4,
          "width": "50%",
        },
      ]
    }
  />
</View>
`;
